### PR TITLE
chore: vercel enable middleware and i18n redirection

### PR DESCRIPTION
**Pull request by ovflowd**
This PR introduces the `middleware` that will take on i18n Language Detection (through the `Accepted-Language` Header.

This is an SSR-exclusive feature and does **not** affect the static builds. When using Static Generation, the Middlewares will be ignored. The codebase is still designed in a way to be entirely static compatible.

This PR also introduces an `optimized` Image Loader (that also requires SSR). This change requires SSR but can easily be disabled by setting the config from `false` to `true` (for the image.unoptimized key).

### CODE REVIEW

**Comment by ovflowd** [comment]
It looks like there's an issue on Vercel's side regarding the failed Storybook deployment (https://github.com/orgs/vercel/discussions/1321)

File: next.config.mjs [hunk 1]
```diff
@@ -20,7 +20,7 @@ const withNextra = nextra({
 
 export default withNextra({
   trailingSlash: false,
-  images: { unoptimized: true },
+  images: { unoptimized: false },
```
Comment (AugustinMauroy) [inline next.config.mjs]: We can just remove this config?

Comment (ovflowd) [reply next.config.mjs]: Rather not, as people should be able to easily turn this setting off.

**Comment by AugustinMauroy** [comment]
Does this code snippet remove `LocalizedLink` and replace it with `Link`?

**Comment by ovflowd** [comment]
> Does this code snippet remove `LocalizedLink` and replace it with `Link`?

No. Otherwise, I would have replaced the Component 😉

**Comment by cb1kenobi** [comment]
Good news! Vercel CLI 29.1.1 has been released with support for Storybook projects! The Storybook framework can now be selected under the project settings. Please update your CLI, then update your project's framework and you should be good to go. Cheers!

**Comment by ovflowd** [comment]
Thank you @cb1kenobi this is fantastic!

**Comment by ovflowd** [comment]
Hey @nodejs/website could I get some reviews here?

File: middleware.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
+import { NextResponse } from 'next/server';
+import localeConfig from './i18n/config.json';
+import type { NextRequest } from 'next/server';
+
+const LOCALE_CODES = localeConfig.map(locale => locale.code);
```
Comment (HinataKah0) [inline middleware.ts]: Nit: Through quick search, it seems that we usually use `camelCase` for constants. I know Google style guide allows `SNAKE_CASE`.

**Review by HinataKah0 — APPROVED** [review]
LGTM!
I tried playing around with [it](https://nodejs-org-git-fork-ovflowd-chore-enable-i18n-routing-openjs.vercel.app/) too, and it seems fine.
Left 1 unblocking nitpick.

**Pull request merged**